### DISCUSSION TITLE
urgent reset the version value

### DISCUSSION THREAD

**Discussion by dnyandeobharambe** [GitTools/GitVersion#3982] [General]
I am using version 5.11.1 tried 5.12 too. I am testing it using my feature branch. I expect my next version should be 1.1.1. But it not consider this and set something like 3082025.2.0+1 . Can I reset this values so that I can use custom for first time using next-version or explicitly tag repo with number. 
I already try to delete tag and all but it did not helped me. Here is my config 
next-version: 1.1.1
major-version-bump-message: '\+semver:\s?(breaking|major)'
minor-version-bump-message: '\+semver:\s?(feature|minor)'
patch-version-bump-message: '\+semver:\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
branches:
  release:
    regex: ^releases?[/-]
    tag: ''
  feature:
    regex: ^feature?[/-]
    tag: ''

Note- I configured it in azure devops and using it in pipeline.
I already trying this from last 3 days. Already post question but not much inputs. If nothing works then I have to scrap it and look alternate library for semantic versioning in my pipeline

**Comment by arturcic** [comment]
Maybe you can try setting the [ignore](https://gitversion.net/docs/reference/configuration#ignore) configuration

```yaml
ignore:
  sha: []
  commits-before: yyyy-MM-ddTHH:mm:ss
```

**Reply by arturcic** [reply]
You can set the `commits-before` to a commit that is the previous commit sha or so and see if that helps

**Reply by dnyandeobharambe** [reply]
Ignore SHA not worked. It created new version again as big number 
tried with commit-before also again big number for major 3082028.0.0
I was expecting 1.0.0 as it is my next-version in config.

next-version: 1.0.0
major-version-bump-message: '\+semver:\s?(breaking|major)'
minor-version-bump-message: '\+semver:\s?(feature|minor)'
patch-version-bump-message: '\+semver:\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
ignore:

I am using this in  azure devops pipeline. Does gitversion hold this value somewhere and refer it when calculate new version?
Is any way to reset it

additionaly this is code I use to read commit message

check_semver_format() {
            # Function definition here
            semver_pattern='\+semver:(breaking|major|feature|minor|fix|patch)'
            if [[ $1 =~ ^$semver_pattern ]]; then
                echo "startsWith"
            else
                if [[ $1 =~ $semver_pattern ]]; then
                    echo "contains"
                else
                    echo "doesNotMatch"
              fi
          fi
          }
          # Example usage
          given_string=$(git log --format=%B -n 1)
          result=$(check_semver_format "$given_string")
          echo "Result: $result"
          # Set a variable for use in subsequent jobs
          echo "##vso[task.setvariable variable=SemverResult;isOutput=true]$result"
        name: RunBashScript

**Reply by dnyandeobharambe** [reply]
I see something this is log , is this tell reason and how I can fix it?
NextVersion in GitVersion configuration file: 1.0.0 with commit count source External Source
    INFO [03/28/24 22:42:04:20] Fallback base version: 0.1.0 with commit count source f74fcc191954fcbe0e4fb581e7ccb9180932b889
    INFO [03/28/24 22:42:04:32] Git tag '3082027.0.0': 3082027.0.0 with commit count source 2822b01e984b65296aa8ff36cbe81e836748949a
    INFO [03/28/24 22:42:04:32] Git tag '3082026.0.0': 3082026.0.0 with commit count source 3ada2dfd8cf6673644979fabbde05391d5cd3aea
    INFO [03/28/24 22:42:04:32] Git tag '3082025.2.0': 3082025.2.0 with commit count source 50bb9b516f92e6967f471248b289e76b2cc205e5
    INFO [03/28/24 22:42:04:32] Git tag '3082025.1.0': 3082025.1.0 with commit count source 003a163ac018b3c6cc7050f6feefb9c1b9998b95
    INFO [03/28/24 22:42:04:32] Git tag '3082025.0.0': 3082025.0.0 with commit count source bef360d130251fdf46203c31b9ec0d92ef1b0b80
    INFO [03/28/24 22:42:04:32] Git tag '3082024.0.0': 3082024.0.0 with commit count source fd73a6a755168499312bfc49d259c84279f1a286

**Reply by dnyandeobharambe** [reply]
I think reset this in git could solve, but I do not know how I can do it any clue?

**Reply by arturcic** [reply]
> I think reset this in git could solve, but I do not know how I can do it any clue?

Seems like you have plenty of tags like '3082027.0.0'. you need to remove all of them, or rename them with a specific prefix

**Reply by dnyandeobharambe** [reply]
I delete all tags, but still it take it and I think reason could be it take commit number of release branch from history. As I saw following in log, I am not sure how to get rid of it , something like pointing to new number forcefully . Is there any way I can add in config (ignore I tried and it did not worked).?

release/branch-to-use-to-merge-from-dev-int-to-main-int-02022023, dev, main
    INFO [04/01/24 20:42:06:72] End: Finding branch source of 'release/hawkeye-release-03082024' (Took: 75.61ms)
    INFO [04/01/24 20:42:06:73] Release branch exists -> Version in branch name: 2022023.0.0 with commit count source 5c38f6d90f1a1e7c760f0be8573effe167b62f51
    INFO [04/01/24 20:42:06:73] Release branch exists -> Version in branch name: 3082024.0.0 with commit count source 219c24033977c239d395700c44de0e6501f7508d
    INFO [04/01/24 20:42:06:81] Found multiple base versions which will produce the same SemVer (3082028.1.0), taking oldest source for commit counting (Git tag '3082028.0.0')
    INFO [04/01/24 20:42:06:81] Base version used: Git tag '3082028.0.0': 3082028.0.0 with commit count source 5cf645c83b8892d5b0758f8482271cb35a905de2
  INFO [04/01/24 20:42:06:81] End: Calculating base versions (Took: 887.42ms)

**Reply by dnyandeobharambe** [reply]
I am able to reset using ignore. But now not able to bump it for patch . This is from log , any clue what is going on? and why it not able to bump patch (I am in feature branch). 

release/branch-to-use-to-merge-from-dev-int-to-main-int-02022023, dev, main
2024-04-03T21:02:41.7430268Z     INFO [04/03/24 21:02:41:55] End: Finding branch source of 'release/hawkeye-release-03082024' (Took: 63.93ms)
2024-04-03T21:02:41.7430767Z     INFO [04/03/24 21:02:41:56] Release branch exists -> Version in branch name: 2022023.0.0 with commit count source 5c38f6d90f1a1e7c760f0be8573effe167b62f51
2024-04-03T21:02:41.7431079Z     INFO [04/03/24 21:02:41:56] Source was ignored due to commit date being outside of configured range
2024-04-03T21:02:41.7431565Z     INFO [04/03/24 21:02:41:56] Release branch exists -> Version in branch name: 3082024.0.0 with commit count source 219c24033977c239d395700c44de0e6501f7508d
2024-04-03T21:02:41.7431892Z     INFO [04/03/24 21:02:41:56] Source was ignored due to commit date being outside of configured range
2024-04-03T21:02:41.7432385Z     INFO [04/03/24 21:02:41:62] Found multiple base versions which will produce the same SemVer (2.1.0), taking oldest source for commit counting (Git tag '2.0.0')
2024-04-03T21:02:41.7432878Z     INFO [04/03/24 21:02:41:62] Base version used: Git tag '2.0.0': 2.0.0 with commit count source 251b1130e5a9fa96d18b9382cf3b359d45238d6c
2024-04-03T21:02:41.7433240Z   INFO [04/03/24 21:02:41:62] End: Calculating base versions (Took: 755.66ms)
2024-04-03T21:02:41.7433503Z   INFO [04/03/24 21:02:41:62] 3 commits found between 251b113 +semver: patch and 820f2dd +semver: patch
2024-04-03T21:02:41.7433786Z   INFO [04/03/24 21:02:41:62] 3 commits found between 251b113 +semver: patch and 820f2dd +semver: patch

**Comment by dnyandeobharambe** [comment]
Just as additional info. I am attaching complete log , so somebody can guide me 
[log-gitversion.txt](https://github.com/GitTools/GitVersion/files/14841463/log-gitversion.txt)

**Reply by arturcic** [reply]
@dnyandeobharambe do you mind cloning the repo and debugging it locally? Maybe you can understand why it's behaving like this?

**Reply by dnyandeobharambe** [reply]
that could be big work as I am directly using it pipeline (no local), before that can you redirect class/method which get call when I calculate version So I can look source code.
Note- This behavior for project branch where lot of merge and release happen without any version in past. If I create fresh small branch and do things it works fine. So might be when it sees release branch as merge it take that commit count or something like that. But why it does I can not understand

**Reply by dnyandeobharambe** [reply]
@arturcic  QQ is there way to ignore all commit except last one? how I can do, I guess I can try that

**Reply by HHobeck** [reply]
I think this is the way you should go:

https://github.com/GitTools/GitVersion/discussions/3982#discussioncomment-8945724

Please try the latest preview release 6.0.0-beta.7.

**Reply by dnyandeobharambe** [reply]
I am using GitTool it add as extenstion in azure devops and it using latest 1.1.1.24031221 (Latest)

**Reply by dnyandeobharambe** [reply]
I did something like this
I explicitly tag as v1.0.0 using git tag .
In config I added it ignore:
  commits-before: 2024-04-03T10:15:00
and next time I saw version as 2.0.0 . It is still wrong (Note I am in feature branch) , I provide comment to bump patch. But It bump major . Is this due to I am in feature branch and not in main? It keep bumping only major even in command if I say patch or minor. 

Here is my complete config for reference
mode: Mainline
major-version-bump-message: '\+semver:\s?(breaking|major)'
minor-version-bump-message: '\+semver:\s?(feature|minor)'
patch-version-bump-message: '\+semver:\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
ignore:
  commits-before: 2024-04-03T10:15:00
branches:
  release:
    regex: ^releases?[/-]
    tag: ''
  feature:
    regex: ^feature?[/-]
    tag: ''
